c_tests/tests/switch.c

```diff
@@ -0,0 +1,41 @@
+// Compiler:
+// Run-time:
+
+// Ensure that an LLVM switch statement is correctly handled.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  int x = 1, res = 0;
+  void *tt = __yktrace_start_tracing(HW_TRACING, &x, &res);
+  NOOPT_VAL(x);
+  switch (x) {
+  case 1:
+    res = 5;
+    break;
+  case 2:
+    res = 12;
+    break;
+  case 3:
+    res = 4;
+    break;
+  default:
+    res += 1;
+  }
```

Comment:
I thought that we were not going to guard the resulting values? Only the inputs?

https://github.com/ykjit/yk/pull/376#discussion_r672232482

(To be honest, I'm more comfortable with guarding both inputs and outputs)

Reply:
You're just making me repeat what I said just below that comment aren't you?

> You're right and I'm wrong

This should mean that I lose face, but since I'm wrong so often, repeating one of my wrongnesses one more time won't hurt me :p

Reply:
Ah, actually I screwed up. I didn't see your "You're right and I'm wrong" comment and *removed* the second guard from the `NOOPT_VAL` test! So I should go back and correct that.

😆

Reply:
Oops, yep!